hoomd/ForceCompute.cc

```diff
@@ -336,26 +343,39 @@ pybind11::object ForceCompute::getEnergiesPython()
         {
         dims[0] = 0;
         }
-    std::vector<double> energy(dims[0]);
+    std::vector<double> global_energy(dims[0]);
 
-    // This is slow: TODO implement a proper gather operation
-    for (unsigned int i = 0; i < m_pdata->getNGlobal(); i++)
+    // sort energies by particle tag
+    std::vector<double> local_energy;
+    local_energy.reserve(m_pdata->getN());
+    std::vector<uint32_t> local_tag(m_pdata->getN());
```

Comment:
Might be worth a helper function to get sorted local tags. We use the same code in each of the functions.